### ISSUE TITLE
Lesson Feedback%3A Verifying Storage on Filecoin - Lesson 3 (Proof-of-Replication (PoRep))

### ISSUE THREAD

**Issue by porcuquine** [ProtoSchool/protoschool.github.io#454]
Have a question or suggestion regarding a specific ProtoSchool lesson? Please use this template to share it!

**URL of the lesson that's confusing:**

https://proto.school/#/verifying-storage-on-filecoin/03

**What's confusing about this lesson?**

> During sealing, the sector data (identified by the CommD) is encoded through a sequence of graph and hashing processes to create a unique replica. The root hash of the resulting replica is the CommR (Commitment of Replication, aka SealedSectorCID), which is recorded to the blockchain.

This is not quite accurate — or, the way in which it is accurate is complicated. I would not describe `CommR` as the root hash of the replica. It is a commitment to the replica, though. If just changing that wording works for you, it works for me. Otherwise, a more elaborate explanation may be required if you want to be accurate in a way that's also not potentially misleading.

**What additional context could we provide to help you succeed?**

**What other feedback would you like to share about ProtoSchool?**

**Comment by terichadbourne** [comment]
Thanks for the feedback @porcuquine! @mishmosh what do you recommend for beginner-friendly rewording here?

As a novice, viewing a CID as "a commitment to a replica" doesn't mean much to me. As an example of the vocab in normal English, I make a commitment to my landlord to pay rent, and the proof that I made that commitment is my lease. Are the CommD and CommR vaguely lease-like in this analogy? Or are they a CID (hash) of the lease? Or neither? 😂

**Comment by porcuquine** [comment]
Regarding normal English vocab: 'commitment' in this sense is not like 'obligation'. It's more like writing a prediction in a sealed envelope and putting the sealed envelope in a public place. Later, when I claim to have predicted the eventual outcome, I open the envelope and show that I had indeed done so. You can trust that it was a real prediction because I committed to it in advance. If I only produce the envelope after the event has occurred, my uncommitted prediction doesn't mean much. As an aside, I would try to distinguish technical terms (in this context) from normal English vocab or things are going to get confusing.

I'm not going to address 'lease' in this context because it either doesn't apply or I don't understand the translation you intend well enough.

CommD *is* the root hash of a merkle tree (of the original data). So if that makes sense to you and your readers, it's fine.

There exists a parallel value, CommRLast — which is also the root hash of a merkle tree (of the replica). If you were talking about CommRLast, where you mention CommR, everything would still make perfect sense.

The problem is that the actual value, CommR, which is what goes on chain, is *not* that root hash. CommR is the hash of *two* other root hashes (CommC and CommR). While you *could* therefore describe CommR as being the 'root hash' of *something*, that something wouldn't be a merkle tree built over the replica. It would instead be something more complicated.

What language you choose to use depends on the level of detail you want to go into. If the goal is to avoid going *any* deeper than already, and to avoid the complexity I presented above, then you might be better off avoiding the semblance of technical terms. If you rephrase this as a 'magical value' which facilitates proofs, no one will mistake that for a precise description and draw the wrong conclusion. (I suspect you don't want to go that far either, so I am just delimiting the range within which the right answer likely lies.)

**Comment by terichadbourne** [comment]
Thanks for this @porcuquine. The envelope concept is helpful for me. 

I totally agree that we should distinguish technical from normal vocab, which is why I raised that misconception. Your original suggestion of changing the wording to "it's a commitment to a replica" without me as a reader already knowing the technical definition of commitment would lead me to wonder what I was promising the replica I'd do for it. 🤦‍♀️

I'll wait and let @mishmosh weigh in on whether we should get into the CommRLast and CommC.

**Comment by mishmosh** [comment]
@porcuquine how would you describe CommC? If we can boil down a description, I propose modifying the lesson like this. This gets fairly detailed, but nowhere else is this detail well-described and I think it's worth including here.

> The root hash of the resulting replica is the CommRLast.

> Next, the CommRLast is hashed together with CommC (_definition of CommC_) to generate the CommR (Commitment of Replication, aka SealedSectorCID), which is recorded to the blockchain.

**Comment by porcuquine** [comment]
It's the merkle root of a tree which facilitates proving that the replica was correctly created from the original data. Any more detail than that will get pretty complicated, so I will provide only if you want more.

**Comment by mishmosh** [comment]
Is this technically correct?

> Next, the CommRLast is hashed together with CommC (another merkle root output from Proof of Replication) to generate the CommR (Commitment of Replication, aka SealedSectorCID), which is recorded to the blockchain.

**Comment by porcuquine** [comment]
I think that works.

**Comment by terichadbourne** [comment]
Have updated to @mishmosh's proposed language in PR #464, with other minor tweaks. Would love your review on that PR @porcuquine.

**Comment by terichadbourne** [comment]
Closed by #464
